src/coreclr/vm/comdynamic.cpp

```diff
@@ -910,5 +887,64 @@ void QCALLTYPE COMDynamicWrite::DefineCustomAttribute(QCall::ModuleHandle pModul
         }
     }
 
+    // Debuggable attribute processing
+    if (((strcmp(szNamespace, FRIEND_ASSEMBLY_TYPE_NAMESPACE) == 0) && (strcmp(szName, FRIEND_ASSEMBLY_TYPE_NAME) == 0)) ||
+        ((strcmp(szNamespace, SUBJECT_ASSEMBLY_TYPE_NAMESPACE) == 0) && (strcmp(szName, SUBJECT_ASSEMBLY_TYPE_NAME) == 0)))
+    {
+        Assembly* pAssembly = pModule->GetAssembly();
+        pAssembly->UpdateCachedFriendAssemblyInfo();
+    }
+}
+
+void QCALLTYPE COMDynamicWrite::DefineCustomAttribute(QCall::ModuleHandle pModule, INT32 token, INT32 conTok, LPCBYTE pBlob, INT32 cbBlob, BOOL toDisk)
+{
+    QCALL_CONTRACT;
+
+    BEGIN_QCALL;
+
+    RefClassWriter* pRCW = pModule->GetReflectionModule()->GetClassWriter();
+    _ASSERTE(pRCW);
+
+    HRESULT hr;
+    mdCustomAttribute retToken;
+
+    if (toDisk && pRCW->GetOnDiskEmitter())
+    {
+        hr = pRCW->GetOnDiskEmitter()->DefineCustomAttribute(
+                token,
+                conTok,
+                pBlob,
+                cbBlob,
+                &retToken);
+    }
+    else
+    {
+        hr = pRCW->GetEmitter()->DefineCustomAttribute(
+                token,
+                conTok,
+                pBlob,
+                cbBlob,
+                &retToken);
+    }
+
+    if (FAILED(hr))
+    {
+        // See if the metadata engine gave us any error information.
```

Comment:
Does the metadata engine ever set IErrorInfo?

(I know that this code was there even before your change.)

Reply:
I'll check. As you say, this code is old and has relics of a bygone era in it..

Reply:
Nope. IErrorInfo isn't set by anything in metadata. I'll get rid of this stuff.